pkg/config/getters.go

```diff
@@ -161,3 +164,144 @@ func GetStoragePath() string {
 func PreferredAddress() string {
 	return os.Getenv("BACALHAU_PREFERRED_ADDRESS")
 }
+
+func GetStringMapString(key string) map[string]string {
+	return viper.GetStringMapString(key)
+}
+
+func Get[T any](key string) (T, error) {
+	raw := viper.Get(key)
+	if raw == nil {
+		return zeroValue[T](), fmt.Errorf("value not found for %s", key)
+	}
+
+	var val T
+	val, ok := raw.(T)
+	if !ok {
+		err := ForKey(key, &val)
```

Comment:
There's a lot of code associated with this path through `Get[T]` do you have an ide of how often we expect to take this path?

Reply:
To answer my own question.  Of the 7 times this func (`Get`) is called, 6 times it is `Get[string]` and once it is `Get[bool]`.